lib/common/xml.c

```diff
@@ -1315,15 +1315,14 @@ write_xml_file(xmlNode * xml_node, const char *filename, gboolean compress)
 
 // Replace a portion of a dynamically allocated string (reallocating memory)
 static char *
-replace_text(char *text, int start, int *length, const char *replace)
+replace_text(char *text, int start, size_t *length, const char *replace)
 {
-    int lpc;
-    int offset = strlen(replace) - 1;   /* We have space for 1 char already */
+    size_t offset = strlen(replace) - 1; // We have space for 1 char already
 
     *length += offset;
     text = pcmk__realloc(text, *length);
 
-    for (lpc = (*length) - 1; lpc > (start + offset); lpc--) {
+    for (size_t lpc = (*length) - 1; lpc > (start + offset); lpc--) {
```

Comment:
Can you guarantee here that *length will be at least 1 the first time through?  If it's 0, lpc will start as some very large value (due to -1 being expressed as an unsigned int) and this loop won't work as you expect.

Reply:
Yes, this is a static function, and used only by crm_xml_escape() in the same file, which initializes length to 1 + strlen(text).

Reply:
And it looks like crm_xml_escape won't call this if text is NULL, so that should be fine.